providers/provider_data.go

```diff
@@ -94,3 +110,116 @@ func defaultURL(u *url.URL, d *url.URL) *url.URL {
 	}
 	return &url.URL{}
 }
+
+// ****************************************************************************
+// These private OIDC helper methods are available to any providers that are
+// OIDC compliant
+// ****************************************************************************
+
+// OIDCClaims is a struct to unmarshal the OIDC claims from an ID Token payload
+type OIDCClaims struct {
+	Subject  string   `json:"sub"`
+	Email    string   `json:"-"`
+	Groups   []string `json:"-"`
+	Verified *bool    `json:"email_verified"`
+
+	raw map[string]interface{}
+}
+
+func (p *ProviderData) verifyIDToken(ctx context.Context, token *oauth2.Token) (*oidc.IDToken, error) {
+	rawIDToken := getIDToken(token)
+	if strings.TrimSpace(rawIDToken) != "" {
+		if p.Verifier == nil {
+			return nil, ErrMissingOIDCVerifier
+		}
+		return p.Verifier.Verify(ctx, rawIDToken)
+	}
+	return nil, ErrMissingIDToken
```

Comment:
Unindent maybe?
```suggestion
	if strings.TrimSpace(rawIDToken) == "" {
		return nil, ErrMissingIDToken
	}
	if p.Verifier == nil {
		return nil, ErrMissingOIDCVerifier
	}
	return p.Verifier.Verify(ctx, rawIDToken)
```

Reply:
Good catch! I'm with you on less nested conditionals, I made these changes and a few more I caught when eyeballing this PR again. Thanks.